stock_operating_unit/model/stock_move_line.py

```diff
@@ -0,0 +1,61 @@
+# © 2020 brain-tec group
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    operating_unit_id = fields.Many2one(
+        "operating.unit",
+        compute="_compute_operating_unit_id",
+        store=True,
+        string="Source Location Operating Unit",
+    )
+
+    operating_unit_dest_id = fields.Many2one(
+        "operating.unit",
+        compute="_compute_operating_unit_dest_id",
+        store=True,
+        string="Dest. Location Operating Unit",
+    )
+
+    @api.depends(
+        "location_id",
+        "location_id.operating_unit_id",
+        "picking_id.picking_type_id",
+        "picking_id.picking_type_id.warehouse_id",
+        "picking_id.picking_type_id.warehouse_id.operating_unit_id",
+    )
+    def _compute_operating_unit_id(self):
+        for rec in self:
+            rec.operating_unit_id = (rec.location_id.operating_unit_id or
+                                     rec.picking_id.picking_type_id.warehouse_id.operating_unit_id)
+
+    @api.depends(
+        "location_id",
+        "location_id.operating_unit_id",
+        "picking_id.picking_type_id",
+        "picking_id.picking_type_id.warehouse_id",
+        "picking_id.picking_type_id.warehouse_id.operating_unit_id",
+    )
+    def _compute_operating_unit_dest_id(self):
+        for rec in self:
+            rec.operating_unit_dest_id = (rec.location_dest_id.operating_unit_id or
+                                          rec.picking_id.picking_type_id.warehouse_id.operating_unit_id)
+
+    @api.constrains("picking_id", "location_id", "location_dest_id")
+    def _check_operating_units(self):
+        for rec in self:
+            ou_pick = rec.picking_id.operating_unit_id or False
+            ou_src = rec.operating_unit_id or False
+            ou_dest = rec.operating_unit_dest_id or False
```

Comment:
The "or False" is not necessary.
You could do direct "if rec.picking_id.operating_unit_id"
The if statement would be true if the field operating_unit_id contains a operating_unit record and otherwise the if statement would be false. You don't need to change it but for future you can save some code.

Reply:
Done correction in the commit [4946d03](https://github.com/brain-tec/operating-unit/pull/343/commits/4946d032ec11f1567f70f9e9c47234e7729c5532)

Reply:
Now it looks good to me!